### PR TITLE
docs: set up the docs/ folder

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR features initial setup of the `docs/` folder for Fleet documentation on the upstream end.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A.

### Special notes for your reviewer

N/A.

### CODE REVIEW

File: docs/howtos/README.md
Comment (zhiying-lin) [inline docs/howtos/README.md:1]: What's the "howtos"? is it "Debugging Guide"? or "Scenarios"?

Comment (michaelawyu) [reply docs/howtos/README.md:1]: It is a very common terms used in technical documentation, usu. reserved for instructions on specific functionalities.

Examples:
In Azure documentation: https://learn.microsoft.com/en-us/azure/aks/scale-cluster?tabs=azure-cli
In GCP documentation: https://cloud.google.com/dataproc/docs/how-to

Comment (michaelawyu) [reply docs/howtos/README.md:1]: Either way's OK with me. We could put in the how-to guides, or make a dedicated folder for it.

**Review by zhiying-lin — DISMISSED** [review]

**Comment by michaelawyu** [comment]
Merged to unblock the progress; approval acquired previously.

If there's any concern, please let me know.

**Pull request merged**